src/data-workspace/has-comment/has-comment-provider.js

```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types'
+import React, { useState, useCallback } from 'react'
+import { getFieldId } from '../get-field-id.js'
+import { HasCommentContext } from './has-comment-context.js'
+
+export const createCommentLookupFromDataSetValues = (dataSetValues) =>
+    Object.values(dataSetValues).reduce((lookup, dataElementCocs) => {
+        for (const [
+            categoryOptionComboId,
+            { dataElement: dataElementId, comment },
+        ] of Object.entries(dataElementCocs)) {
+            if (comment) {
+                lookup.add(getFieldId(dataElementId, categoryOptionComboId))
+            }
+        }
+        return lookup
+    }, new Set())
+
+export default function HasCommentProvider({ children }) {
```

Comment:
Should we add a TODO or a comment about wanting to investigate better query-keys and how that might mitigate the use of this context?

Reply:
No we should not. Please see [this Slack thread](https://dhis2.slack.com/archives/CJF3MDK8Q/p1661859412925369) for some additional info. In the thread I suggest writing a JIRA issue, but in this case we should first actually discuss within the team how we would like to improve/refactor this app, so a JIRA ticket isn't very suitable yet. Instead I have created #133 where we can discuss improvements. Once we finalise that discussion we can create a JIRA issue/epic for it. And then during the v2.40 cycle we can see if we can allocate some time to work on it.